src/main.ts

```diff
@@ -82,14 +83,15 @@ async function _confirmPassword(password: string) {
 /**
  *
  */
-function getDialog(): Vue {
-	if (window._nc_password_confirmation_dialog === undefined) {
-		console.debug('Prompting password form')
-		const dialog = spawnDialog(PasswordDialogVue, {}, () => {})
-		window._nc_password_confirmation_dialog = dialog
+function getDialog(validate: (password: string) => Promise<void>): { dialog: App, eventBus: Emitter<PasswordDialogEvents> } {
+	console.debug('Prompting password form')
+	const eventBus = mitt<PasswordDialogEvents>()
+	const dialog = spawnDialog(PasswordDialogVue, { eventBus, validate }, () => {})
```

Comment:
We don't need an event bus here to handle closing.

If we removed `_nc_password_confirmation_dialog`, we can just pass `onClose` handler here directly as we are passing props.

If we want to reuse the same dialog, we can implement the prompt as the dialog's exposed public method.

Reply:
I talked to Louis and he confirmed that we can drop the global dialog handle.

This simplifies the whole logic and would in theory open up the possibility to use the same code for both Vue versions (vue-demi?).

Reply:
> This simplifies the whole logic and would in theory open up the possibility to use the same code for both Vue versions (vue-demi?).

No. `vue-demi` works well for pure vue libraries, but here we have many dependencies, and they are not compatible with Vue in a "demi" way. Every dependency must also be based on vue-demi, or not use anything changed in Vue 3

Reply:
To support both versions in practice, we can bundle Vue in this library. Then it doesn't matter, what Vue version a library user uses.

However, it has a drawback - it makes the library MUCH bigger in the bundle size. I'll check if we can make it not so big. But it will be several times bigger than the current size.

- https://github.com/nextcloud-libraries/nextcloud-password-confirmation/pull/915

Reply:
We have a `spawnDialog` function no in the `nextcloud-vue`.

https://github.com/nextcloud-libraries/nextcloud-vue/blob/main/src/functions/dialog/index.ts

Once it's backported to `nextcloud-vue@8`, we can use it in this library. Then the only difference between Vue 2 and Vue 3 versions is dependency versions.